agent/consul/state/events_test.go

```diff
@@ -0,0 +1,148 @@
+				Index:   1,
+			},
+			err: nil,
+		},
+		"Deprecated top level fields": {
+			req: &pbsubscribe.SubscribeRequest{
+				Topic:     EventTopicServiceDefaults,
+				Key:       "key",
+				Partition: "partition",
+				Namespace: "consul",
+				PeerName:  "peer",
+			},
+			entMeta: acl.EnterpriseMeta{},
+			expectedSubscribeRequest: &stream.SubscribeRequest{
+				Topic: EventTopicServiceDefaults,
+				Subject: EventSubjectConfigEntry{
+					Name:           "key",
+					EnterpriseMeta: &acl.EnterpriseMeta{},
+				},
+			},
+			err: nil,
+		},
+		"Service health": {
+			req: &pbsubscribe.SubscribeRequest{
+				Topic: EventTopicServiceHealth,
+				Subject: &pbsubscribe.SubscribeRequest_NamedSubject{
+					NamedSubject: &pbsubscribe.NamedSubject{
+						Key:       "key",
+						Namespace: "consul",
+						Partition: "partition",
+						PeerName:  "peer",
+					},
+				},
+				Token: aclToken,
+				Index: 2,
+			},
+			entMeta: acl.EnterpriseMeta{},
+			expectedSubscribeRequest: &stream.SubscribeRequest{
+				Topic: EventTopicServiceHealth,
+				Subject: EventSubjectService{
+					Key:            "key",
+					EnterpriseMeta: acl.EnterpriseMeta{},
+					PeerName:       "peer",
+				},
+				Token: aclToken,
+				Index: 2,
+			},
+			err: nil,
+		},
+		"Config": {
+			req: &pbsubscribe.SubscribeRequest{
+				Topic: EventTopicAPIGateway,
+				Subject: &pbsubscribe.SubscribeRequest_NamedSubject{
+					NamedSubject: &pbsubscribe.NamedSubject{
+						Key:       "key",
+						Namespace: "consul",
+						Partition: "partition",
+						PeerName:  "peer",
+					},
+				},
+				Token: aclToken,
+				Index: 2,
+			},
+			entMeta: acl.EnterpriseMeta{},
+			expectedSubscribeRequest: &stream.SubscribeRequest{
+				Topic: EventTopicAPIGateway,
+				Subject: EventSubjectConfigEntry{
+					Name:           "key",
+					EnterpriseMeta: &acl.EnterpriseMeta{},
+				},
+				Token: aclToken,
+				Index: 2,
+			},
+			err: nil,
+		},
+		"Service list without wildcard returns error": {
+			req: &pbsubscribe.SubscribeRequest{
+				Topic: EventTopicServiceList,
+				Subject: &pbsubscribe.SubscribeRequest_NamedSubject{
+					NamedSubject: &pbsubscribe.NamedSubject{
+						Key:       "key",
+						Namespace: "consul",
+						Partition: "partition",
+						PeerName:  "peer",
+					},
+				},
+			},
+			entMeta:                  acl.EnterpriseMeta{},
+			expectedSubscribeRequest: nil,
+			err:                      fmt.Errorf("topic %s can only be consumed using WildcardSubject", EventTopicServiceList),
+		},
+		"Unrecognized topic returns error": {
+			req: &pbsubscribe.SubscribeRequest{
+				Topic: 99999,
+			},
+			entMeta:                  acl.EnterpriseMeta{},
+			expectedSubscribeRequest: nil,
+			err:                      fmt.Errorf("cannot construct subject for topic 99999"),
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			actual, err := PBToStreamSubscribeRequest(tc.req, tc.entMeta)
+
+			if tc.err != nil {
+				require.Error(t, err)
```

Comment:
Is the intention here to see if the error matches? If so, we could
```go
require.EqualError(t, err, tc.err.Error())
```

Reply:
Awesome! I even caught a bug in my test with this change. Thanks!